rust-runtime/aws-smithy-mocks-experimental/tests/get-object-mocks.rs

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::config::Region;
+use aws_sdk_s3::operation::get_object::{GetObjectError, GetObjectOutput};
+use aws_sdk_s3::operation::list_buckets::ListBucketsError;
+use aws_sdk_s3::{Client, Config};
+use aws_smithy_types::body::SdkBody;
+use aws_smithy_types::byte_stream::ByteStream;
+use aws_smithy_types::error::metadata::ProvideErrorMetadata;
+use aws_smithy_types::error::ErrorMetadata;
+
+use aws_smithy_mocks_experimental::{mock, MockResponseInterceptor};
+
+const S3_NO_SUCH_KEY: &'static str = r#"<?xml version="1.0" encoding="UTF-8"?>
+<Error>
+  <Code>NoSuchKey</Code>
+  <Message>The resource you requested does not exist</Message>
+  <Resource>/mybucket/myfoto.jpg</Resource>
+  <RequestId>4442587FB7D0A2F9</RequestId>
+</Error>"#;
+
+#[tokio::test]
+async fn create_mock_s3_get_object() {
+    let s3_404 = mock!(Client::get_object)
+        .match_requests(|inp| {
+            inp.bucket() == Some("test-bucket") && inp.key() != Some("correct-key")
+        })
+        .then_http_response(|| {
+            http::Response::builder()
+                .status(400)
+                .body(SdkBody::from(S3_NO_SUCH_KEY))
+                .unwrap()
+                .try_into()
+                .unwrap()
+        });
+
+    let s3_real_object = mock!(Client::get_object)
+        .match_requests(|inp| {
+            inp.bucket() == Some("test-bucket") && inp.key() == Some("correct-key")
+        })
+        .then_output(|| {
+            GetObjectOutput::builder()
+                .body(ByteStream::from_static(b"test-test-test"))
+                .build()
+        });
+
+    let modeled_error = mock!(Client::list_buckets).then_error(|| {
+        ListBucketsError::generic(ErrorMetadata::builder().code("InvalidAccessKey").build())
```

Comment:
`ListBucketsError` implements `ProvideErrorMetadata`. Given that, would it be possible to translate it into this `ErrorMetadata` struct rather than having to manually construct it?

Reply:
I don't think I follow

Reply:
oh if there are modeled errors, you can construct them directly

Reply:
Should just be reflected in docs/examples then.

Reply:
done—see `s3-getobject-mocks.rs`